mbed_host_tests/host_tests_runner/mbed_base.py

```diff
@@ -53,6 +53,8 @@ def __init__(self, options):
         self.serial_baud = DEFAULT_BAUD_RATE
         self.serial_timeout = 1
 
+        self.tags = self.options.allocate_on_tag
```

Comment:
Is this assignment really needed? why not just use `self.options.allocate_on_tag` directly in place of `self.tags`?  Further, you should ensure that if an object is constructed, it has a valid `tags` attribute, so that you don't have to do odd `getattr` shenanigans later.

Reply:
The other components of the config structure from [here](https://github.com/ARMmbed/htrun/blob/master/mbed_host_tests/host_tests_runner/host_test_default.py#L162) are defined above, the options structure that contains the tags at this level are not passed through the config structure [here](https://github.com/ARMmbed/htrun/pull/163/files#diff-ab29c0280949087d36c3c6cca0af4441R23).

Was that what you were referring to? I was not able to find another way to pass the existing options.allocate_on_tag data to the underlying RemoteConnectorPrimitive constructor.